File: libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/route/RouteListener.java
package org.maplibre.navigation.android.navigation.v5.route;

import androidx.annotation.Nullable;

import org.maplibre.navigation.android.navigation.v5.models.DirectionsResponse;
import org.maplibre.navigation.android.navigation.v5.routeprogress.RouteProgress;

/**
 * Will fire when either a successful / failed response is received.
 */
public interface RouteListener {

  void onResponseReceived(DirectionsResponse response, @Nullable RouteProgress routeProgress);

  void onErrorReceived(Throwable throwable);
}

File: libandroid-navigation/src/main/java/org/maplibre/navigation/android/navigation/v5/route/RouteListener.kt
package org.maplibre.navigation.android.navigation.v5.route

import org.maplibre.navigation.android.navigation.v5.models.DirectionsResponse
import org.maplibre.navigation.android.navigation.v5.routeprogress.RouteProgress

/**
 * Will fire when either a successful / failed response is received.
 */
interface RouteListener {
    fun onResponseReceived(response: DirectionsResponse, routeProgress: RouteProgress)

    fun onErrorReceived(throwable: Throwable)
}
